stix2/utils.py

```diff
@@ -165,46 +165,88 @@ def _get_dict(data):
             raise ValueError("Cannot convert '%s' to dictionary." % str(data))
 
 
+def _iterate_over_values(dict_values, tuple_to_find):
+    """Loop recursively over dictionary values"""
+    from .base import _STIXBase
+    for pv in dict_values:
+        if isinstance(pv, list):
+            for item in pv:
+                if isinstance(item, _STIXBase):
+                    val = find_property_index(
+                        item,
+                        item.object_properties(),
+                        tuple_to_find
+                    )
+                    if val is not None:
+                        return val
```

Comment:
If this is to be a value-returning function, make sure all code paths return a value.  I think it would return None if control dropped off the end of the function, but still.  There are a lot of conditional returns without an "else" clause.

In fact, since None would be returned if there were no explicit "return", code like:
```python
if val is not None:
    return val
# else, don't return anything
```
means
```python
if val is not None:
    return val
else:
    return None
```
which is exactly the same as:
```python
return val
```
The latter is equivalent, simplest, and ensure there's always an explicit "return".
(Apologies if this is still a work in progress and you just hadn't got around to addressing this yet... :))

Reply:
@chisholm, some of the if/else clauses are required because (say a list for example) maybe the first item is not the one I want, so the method would return None. Now on further iteration I can find the object I am searching for and the return a valid index.

Reply:
I agree, @emmanvg. In the remaining cases, the `else` is not `return None` but rather "continue to the next item in the list", so I think we're good on this front!

Reply:
Yeah, I guess I don't know what to return in those cases. My best approach would be to just return something (like a 0) and log it before just to not fail-hard, but that depends if that makes sense as a "solution".